xl2times/transforms.py

```diff
@@ -1475,6 +1511,68 @@ def generate_dummy_processes(
     return tables
 
 
+def process_tradelinks(
+    config: datatypes.Config,
+    tables: List[datatypes.EmbeddedXlTable],
+    model: datatypes.TimesModel,
+) -> List[datatypes.EmbeddedXlTable]:
+    """
+    Transform tradelinks table
+    """
+
+    result = []
+    for table in tables:
+        if table.tag == datatypes.Tag.tradelinks:
+            df = table.dataframe
+            comm = df.columns[0]
+            destinations = list(df.columns).remove(comm)
+            df.rename(columns={comm: "origin"}, inplace=True)
+            df = pd.melt(
+                df, id_vars=["origin"], value_vars=destinations, var_name="destination"
+            )
+            df = df[df["value"] == 1].drop(columns=["value"])
+            df["destination"] = df["destination"].str.upper()
+            df.drop_duplicates(keep="first", inplace=True)
+            df["tradelink"] = 1
+            # Add a column containing linked regions (directionless)
+            df["regions"] = df.apply(
+                lambda row: tuple(sorted([row["origin"], row["destination"]])), axis=1
+            )
+            # Determine whether a trade link is bi- or unidirectional
+            td_type = df.groupby(["regions"])["tradelink"].agg("count").reset_index()
+            td_type.replace({"tradelink": {1: "u", 2: "b"}}, inplace=True)
+            df.drop(columns=["tradelink"], inplace=True)
+            df = df.merge(td_type, how="inner", on="regions")
+            # Drop tradelink (bidirectional) duplicates
+            df.drop_duplicates(subset=["regions"], keep="last", inplace=True)
```

Comment:
It's funny that we drop bidirectional links here but reconstruct them in `generate_trade` :)

Reply:
Yes, that's because we want to get rid of the same bidirectional tradelinks that have different process names. Otherwise we would get 4 tradelinks in `generate_trade` instead of 2. :-)

Reply:
That's only if you put a `b` in the `tradelinks` column. :)

I was wondering if it's possible to simplify the code and leave both directions of links here, if it allows removing the code in `generate_trade` that "doubles" the rows with a `b`?

Reply:
Well, if we don't put b in the column, the links will be processed as unidirectional. :-) This may have an impact on how some other attributes / constraints are applied.

Reply:
> I was wondering if it's possible to simplify the code and leave both directions of links here, if it allows removing the code in `generate_trade` that "doubles" the rows with a `b`?

Not that I see...